### DISCUSSION TITLE
Are PSP for marketplace on the roadmap ?

### DISCUSSION THREAD

**Discussion by ldng** [juspay/hyperswitch#2819] [Ideas / Feature Requests]
Correct me if I am wrong, Hyperswitch seem geared toward ecommerce (which make perfect sense).
Do you plan to also be able at some point to handle marketplace payment workflows ?

In a marketplace setup
- either the marketplace operator has a master account at the PSP that is used to process main payment, and once the money in the master account, you can ask for the money to be ventilated to merchant account;

- or you have an operator/administrative account that allows you to create merchant accounts in the PSP and then you send the main transaction info with sub-payment info and the PSP handle the capture directly in corresponding merchant accounts.

**Comment by manojradhakrishnan** [comment]
Hi @ldng 

We currently support the former solution (which you have proposed above) for e-commerce marketplaces.
* Marketplace operator can accept payins on behalf of merchants into the master account; payins supported across 50+ processors
* Payouts API are available to push the money from master account to merchant bank account or card; payouts are supported via Adyen and Wise. Refer here for more details on payouts: https://hyperswitch.io/docs/features/payouts

While Payins are fully open sourced, the Payout APIs are still in `beta` and hence not open sourced yet. 

If you are looking to use Hyperswitch for such a solution, please let us know your priorities and requirements. We can add it to the roadmap later this quarter.

**Reply by euluizaraujo** [reply]
@manojradhakrishnan Hello, just to update this thread. The payouts API is already 100% open source, correct? Additionally it looks like stripe connect was also added, right?

**Reply by manojradhakrishnan** [reply]
It is still being migrated to open source. We will announce it very soon.

**Reply by euluizaraujo** [reply]
@manojradhakrishnan  Perfect, is there already an estimate?

**Reply by manojradhakrishnan** [reply]
Tentatively by next week. 

Apologies for not being very accurate. Because as you might see our Github activity there are lot of stuff going on simultaneously.

**Reply by euluizaraujo** [reply]
@manojradhakrishnan No problem, thanks for the feedback.  I will follow the updates throughout the week.
